src/main/java/org/spongepowered/common/data/manipulator/immutable/tileentity/ImmutableSpongeBeaconData.java

```diff
@@ -0,0 +1,124 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.manipulator.immutable.tileentity;
+
+import com.google.common.collect.ComparisonChain;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.tileentity.ImmutableBeaconData;
+import org.spongepowered.api.data.manipulator.mutable.tileentity.BeaconData;
+import org.spongepowered.api.data.value.immutable.ImmutableOptionalValue;
+import org.spongepowered.api.effect.potion.PotionEffectType;
+import org.spongepowered.common.data.manipulator.immutable.common.AbstractImmutableData;
+import org.spongepowered.common.data.manipulator.mutable.tileentity.SpongeBeaconData;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeOptionalValue;
+
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+public class ImmutableSpongeBeaconData extends AbstractImmutableData<ImmutableBeaconData, BeaconData> implements ImmutableBeaconData {
+
+    @Nullable
+    final PotionEffectType primaryEffect;
+    @Nullable
+    final PotionEffectType secondaryEffect;
+    final ImmutableOptionalValue<PotionEffectType> primaryEffectValue;
+    final ImmutableOptionalValue<PotionEffectType> secondaryEffectValue;
+
+    public ImmutableSpongeBeaconData(@Nullable PotionEffectType primaryEffect, @Nullable PotionEffectType secondaryEffect) {
+        super(ImmutableBeaconData.class);
+        this.primaryEffect = primaryEffect;
+        this.secondaryEffect = secondaryEffect;
+        this.primaryEffectValue = new ImmutableSpongeOptionalValue<>(Keys.BEACON_PRIMARY_EFFECT, Optional.ofNullable(this.primaryEffect));
+        this.secondaryEffectValue = new ImmutableSpongeOptionalValue<>(Keys.BEACON_SECONDARY_EFFECT, Optional.ofNullable(this.secondaryEffect));
+        registerGetters();
+    }
+
+    @Override
+    protected void registerGetters() {
+        registerFieldGetter(Keys.BEACON_PRIMARY_EFFECT, this::getPrimaryEffect);
+        registerKeyValue(Keys.BEACON_PRIMARY_EFFECT, this::primaryEffect);
+
+        registerFieldGetter(Keys.BEACON_SECONDARY_EFFECT, this::getSecondaryEffect);
+        registerKeyValue(Keys.BEACON_SECONDARY_EFFECT, this::secondaryEffect);
+    }
+
+    public Optional<PotionEffectType> getPrimaryEffect() {
+        return Optional.ofNullable(this.primaryEffect);
+    }
+
+    @Override
+    public ImmutableOptionalValue<PotionEffectType> primaryEffect() {
+        return primaryEffectValue;
+    }
+
+    public Optional<PotionEffectType> getSecondaryEffect() {
+        return Optional.ofNullable(this.secondaryEffect);
+    }
+
+    @Override
+    public ImmutableOptionalValue<PotionEffectType> secondaryEffect() {
+        return secondaryEffectValue;
+    }
+
+    @Override
+    public ImmutableBeaconData clearEffects() {
+        return this;
```

Comment:
This should return a new `ImmutableBeaconData`, with the potion effects cleared.